### PR TITLE
Fix for #718

### PR DESCRIPTION
**Pull request by eboadom**
To close https://github.com/aave/aave-v3-core/issues/718

### CODE REVIEW

File: contracts/protocol/libraries/logic/ReserveLogic.sol [hunk 1]
```diff
@@ -296,19 +298,18 @@ library ReserveLogic {
...
-        uint256 cumulatedVariableBorrowInterest = MathUtils.calculateCompoundedInterest(
-          reserveCache.currVariableBorrowRate,
-          reserveCache.reserveLastUpdateTimestamp
-        );
-        reserveCache.nextVariableBorrowIndex = cumulatedVariableBorrowInterest.rayMul(
-          reserveCache.currVariableBorrowIndex
-        );
-        reserve.variableBorrowIndex = reserveCache.nextVariableBorrowIndex.toUint128();
-      }
+    // Variable borrow side only gets updated if there is any accrual of variable debt
```
Comment (miguelmtzinf) [inline contracts/protocol/libraries/logic/ReserveLogic.sol:303]: This comment seems untrue to me. You are using `currScaledVariableDebt` which is the total variable debt rather than the accrual

**Review by miguelmtzinf — CHANGES_REQUESTED** [review]

File: contracts/protocol/libraries/logic/ReserveLogic.sol (hunk 1 above)
Comment (eboadom) [reply contracts/protocol/libraries/logic/ReserveLogic.sol]: Agree, more precise now here https://github.com/aave/aave-v3-core/pull/724/commits/efdaf634a7809e254ae27c2543f3c64ac875d83e

File: test-suites/pool-edge.spec.ts [hunk 2]
```diff
@@ -793,4 +847,110 @@ makeSuite('Pool: Edge cases', (testEnv: TestEnv) => {
...
+
+    const reserveDataAfter1 = await pool.getReserveData(dai.address);
+
+    expect(reserveDataAfter1.variableBorrowIndex).to.be.eq(reserveDataBefore.variableBorrowIndex);
+    expect(reserveDataAfter1.accruedToTreasury).to.be.gt(reserveDataBefore.accruedToTreasury);
+    expect(reserveDataAfter1.liquidityIndex).to.be.gt(reserveDataBefore.liquidityIndex);
+
+    await advanceTimeAndBlock(10000);
+
+    // "Clean" update, that should not increase the liquidity index, only variable borrow
```
Comment (miguelmtzinf) [inline test-suites/pool-edge.spec.ts:940]:
```suggestion
    // "Clean" update, that should not increase the liquidity index, only stable borrow
```

**Review by miguelmtzinf — APPROVED** [review]
